### PR TITLE
Phan logger syslog

### PR DESCRIPTION
**Pull request by hanpeter**
1. Created `--no-syslog-facility` to disable syslog logging.
1. Modified `setup()` method of `krux.cli` unit test
1. Created unit tests for `--syslog-facility` and `--no-syslog-facility` CLI arguments
1. Created basic unit tests for `krux.logging`

### CODE REVIEW

**Comment by jib** [comment]
Sure this should be on by default, with logs going to both places?

--
Jos Boumans
VP of Operations & Infrastructure
Sent from my iPhone

On May 4, 2016, at 17:42, Peter Han <notifications@github.com<mailto:notifications@github.com>> wrote:


  1.  Created --no-syslog-facility to disable syslog logging.
  2.  Modified setup() method of krux.cli unit test
  3.  Created unit tests for --syslog-facility and --no-syslog-facility CLI arguments
  4.  Created basic unit tests for krux.logging

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/krux/python-krux-stdlib/pull/35

Commit Summary

  *   Added --no-syslog-facility CLI option
  *   Unit test for --syslog-facility CLI arg
  *   Clean up the parser mocking for easier unit test writing
  *   Unit test for --no-syslog-facility CLI arg
  *   Minor bug fix and removing debug code
  *   More unit tests for krux.logging

File Changes

  *   M krux/cli.py<https://github.com/krux/python-krux-stdlib/pull/35/files#diff-0> (12)
  *   M tests/unit_tests/test_cli.py<https://github.com/krux/python-krux-stdlib/pull/35/files#diff-1> (68)
  *   M tests/unit_tests/test_logging.py<https://github.com/krux/python-krux-stdlib/pull/35/files#diff-2> (22)

Patch Links:

  *   https://github.com/krux/python-krux-stdlib/pull/35.patch
  *   https://github.com/krux/python-krux-stdlib/pull/35.diff

File: krux/cli.py [hunk 1]
```diff
@@ -335,6 +334,15 @@ def add_logging_args(parser, stdout_default=True):
...
         '(default: %(default)s).'
     )
+
+    group.add_argument(
+        '--no-syslog-facility',
+        dest='syslog_facility',
+        action='store_const',
+        default=DEFAULT_LOG_FACILITY,
+        const=None,
+        help='disable syslog facility',
```
Comment (paul-krohn) [inline krux/cli.py]: nit: I think you mean 'disable syslog'? maybe call the cli option --no-syslog?

**Comment by paul-krohn** [comment]
@jib It's been set up like this for a while, so if we change it, and want to keep current outputs, we'll need to change calling code.

**Comment by paul-krohn** [comment]
👍 Thank you!

**Comment by hanpeter** [comment]
@jib, what @paul-krohn said. And also, I didn't want to change the existing behaviour. Just wanted to give it a new feature.

**Pull request merged**